internal/sub_reconcilers/reconcile_deployment_updates.go

```diff
@@ -239,10 +243,17 @@ func getPodNamesandStatuses(pods []corev1.Pod) []string {
 func getPodstatusReason(pods []corev1.Pod) float64 {
 	// return 0 if not found, otherwise return 1
 	for _, pod := range pods {
-		if string(pod.Status.Phase) == "Pending" {
-			if string(pod.Status.ContainerStatuses[0].State.Waiting.Reason) == "ErrImagePull" ||
-				string(pod.Status.ContainerStatuses[0].State.Waiting.Reason) == "ImagePullBackOff" {
-				return 1
+		pendingState := string(pod.Status.Phase)
+		if pendingState == "Pending" {
+			if len(pod.Status.ContainerStatuses) > 0 {
+				if len(pod.Status.ContainerStatuses[0].State.Waiting.Reason) > 0 {
+					errorImagePull := string(pod.Status.ContainerStatuses[0].State.Waiting.Reason)
+					imagePullBackOff := string(pod.Status.ContainerStatuses[0].State.Waiting.Reason)
+					if errorImagePull == "ErrImagePull" ||
+						imagePullBackOff == "ImagePullBackOff" {
+						return 1
```

Comment:
does this have to be a float64 ?

Reply:
I think so, prometheus.NewGauge returns float64